drivers/platform/stm32/stm32_pwm.h

```diff
@@ -39,7 +39,10 @@
 #include "stm32_gpio.h"
 #include "stm32_hal.h"
 
-#include <stdbool.h>
+enum stm32_pwm_timer {
+	PWM_TIM_TIMER = 0,
```

Comment:
this needs to be prefixed, I would suggest:
```
enum stm32_pwm_timer {
	STM32_PWM_TIMER_TIM = 0,
	STM32_PWM_TIMER_LPTIM,
}
```

Reply:
Normally I'd use a boolean but it seems there's a 3rd type available in STM32 HAL, HRTIM_HandleTypeDef, that someone may wanna add in the future so enum is ok